core/bentley/src/StatusCategory.ts

```diff
@@ -0,0 +1,355 @@
+  BentleyError,
+  BentleyStatus,
+  BriefcaseStatus,
+  ChangeSetStatus,
+  GeoServiceStatus,
+  HttpStatus,
+  IModelHubStatus,
+  IModelStatus,
+  RealityDataStatus,
+  RepositoryStatus,
+  RpcInterfaceStatus,
+} from "./BentleyError";
+
+/* eslint-disable @typescript-eslint/no-shadow */
+
+/** @alpha */
+export type StatusCategoryHandler = (error: BentleyError) => StatusCategory | undefined;
+
+/** A group of related statuses for aggregate reporting purposes.
+ * @alpha
+ */
+export abstract class StatusCategory {
+  public static handlers: Set<StatusCategoryHandler> = new Set();
+
+  public static for(error: BentleyError): StatusCategory {
+    for (const handler of this.handlers) {
+      const category = handler(error);
+      if (category) {
+        return category;
+      }
+    }
+
+    return lookupCategory(error);
+  }
+
+  public abstract name: string;
+  public abstract code: number;
+  public abstract error: boolean;
+}
+
+/***
+ * A success status.
+ * @alpha
+ */
+export abstract class SuccessCategory extends StatusCategory {
+  public error = false;
+}
+
+/**
+ * An error status.
+ * @alpha
+ */
+export abstract class ErrorCategory extends StatusCategory {
+  public error = true;
+}
+
+namespace HTTP {
+  export class OK extends SuccessCategory { public name = "OK"; public code = 200; }
+  export class Accepted extends SuccessCategory { public name = "Accepted"; public code = 202; }
+  export class NoContent extends SuccessCategory { public name = "NoContent"; public code = 204; }
+
+  export class BadRequest extends ErrorCategory { public name = "BadRequest"; public code = 400; }
+  export class Unauthorized extends ErrorCategory { public name = "Unauthorized"; public code = 401; }
+  export class Forbidden extends ErrorCategory { public name = "Forbidden"; public code = 403; }
+  export class NotFound extends ErrorCategory { public name = "NotFound"; public code = 404; }
+  export class MethodNotAllowed extends ErrorCategory { public name = "MethodNotAllowed"; public code = 405; }
+  export class NotAcceptable extends ErrorCategory { public name = "NotAcceptable"; public code = 406; }
+  export class RequestTimeout extends ErrorCategory { public name = "RequestTimeout"; public code = 408; }
+  export class Conflict extends ErrorCategory { public name = "Conflict"; public code = 409; }
+  export class Gone extends ErrorCategory { public name = "Gone"; public code = 410; }
+  export class PreconditionFailed extends ErrorCategory { public name = "PreconditionFailed"; public code = 412; }
+  export class ExpectationFailed extends ErrorCategory { public name = "ExpectationFailed"; public code = 417; }
+  export class MisdirectedRequest extends ErrorCategory { public name = "MisdirectedRequest"; public code = 421; }
+  export class UnprocessableEntity extends ErrorCategory { public name = "UnprocessableEntity"; public code = 422; }
+  export class Locked extends ErrorCategory { public name = "Locked"; public code = 423; }
```

Comment:
I don't see 423, or 424 documented on MDN, maybe we should avoid them if they're poorly supported?

Reply:
I took them out...